rfcs/quick-start-book.md

```diff
@@ -0,0 +1,352 @@
+4. Examples must pass Bevy's standard `clippy` lints.
+5. The polish level of your examples should correspond to the point you're trying to make.
+   1. If you're demonstrating a new feature, show only the most basic syntax as locally as possible.
+   2. When trying to explain how a game can be made, organize and polish your code to showcase best practices.
+   3. Lack of polish should serve an end: don't show bad or sloppy practices without a good reason.
+   4. Showing how (and why!) to refactor your code is a very powerful teaching tool.
+6. Stick to a consistent style (e.g. for loops vs map) within each example.
+7. If you need to give advice that will only matter to some of your audience (e.g. how to handle an edge case, or support a specific platform), do so in a clearly marked aside or list.
+8. Examples should not use or rely on third-party plugins.
+These may be appropriate to link in "next steps" however at the end of the examples.
+   1. Third-party crates should be limited to the most essential, such as `rand`.
+
+## Implementation strategy
+
+1. The user-facing explanation above should live in `CONTRIBUTING.md` of the [`bevy-website` repo](https://github.com/bevyengine/bevy-website).
+2. This should be linked to from the website, likely under a "Contributing" tab.
+3. Per [bevy-website #143](https://github.com/bevyengine/bevy-website/issues/143),the website will need to be revamped somewhat.
+4. Automatic compilation of code snippets is the critical feature from the above, ensuring that we can keep the examples current in a reliable way.
+5. Once this RFC is merged, work should begin on a separate branch of `bevy-website` repo to create a new book.
+6. For the launch of 0.6 (or perhaps sooner), this branch is merged and becomes the new `main`.
+7. A persistent `bevy-main` branch of the repo is maintained and with a dependency on the `main` branch of `bevy` to ensure that we can test out new changes and refine the code before new features and breaking changes go live and reduce release crunch.
+
+### Quick Start examples
+
+### Breakout
+
+**Genres:** arcade
+**Focus:** basic physics, keyboard controls, sprites
+
+Control a paddle with your keyboard to smash bricks and score points in this classic arcade game!
+
+### Falling Sand
+
+**Genres:** simulation, RTS, factory-builder
+**Focus:** advanced ECS patterns, pixel graphics, handling mouse input
+
+Simulate the interactions between different materials as you build your own creation.
+
+### 3D Puzzle Game
+
+**Genres:** puzzle, 3rd person platformer
+**Focus:** camera controls, rule checking, level creation, 3D graphics
+
+Create devious puzzles for your players to solve as they navigate a 3D world.
+
+### Bevy Book chapters
+
+Advanced chapters are skippable, rely on prior knowledge and are not useful for beginners.
+
+1. Getting Started
+   1. Installing Rust and Bevy
+   2. `App`, `AppBuilder` and `World`
```

Comment:
Maybe "Getting Started" should have quick high level summaries of every major pillar, the general structure of a game, etc.

I like the "category-based structure" we moved to, but the biggest downside is that it doesn't flow as well / might fail to introduce people to concepts fast enough.

"Getting Started" feels like our chance to "sell" people on Bevy: the power of our app model (ECS, plugins / modularity, performance), the features we offer (UI, assets, scenes, audio, etc), open source / community effort, platform support, design philosophy, etc.

Imo people should leave Quick Start:

* With a running empty bevy app
* High level understanding of engine and what it offers
* Jazzed about Bevy

Reply:
Sounds good; I'll think about how to extend this. I'd like to avoid fully duplicating a Quick Start guide though.